src/Draco.LanguageServer/Capabilities/SignatureHelp.cs

```diff
@@ -0,0 +1,23 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Draco.Compiler.Api.CodeCompletion;
+using Draco.Lsp.Model;
+using Draco.Lsp.Server.Language;
+
+namespace Draco.LanguageServer;
+
+internal sealed partial class DracoLanguageServer : ISignatureHelp
+{
+    public SignatureHelpRegistrationOptions SignatureHelpRegistrationOptions => new()
+    {
+        DocumentSelector = this.DocumentSelector,
+        TriggerCharacters = new[] { "(" },
+        RetriggerCharacters = new[] { "," },
+    };
+
+    public Task<SignatureHelp?> SignatureHelpAsync(SignatureHelpParams param, CancellationToken cancellationToken)
+    {
+        var cursorPosition = Translator.ToCompiler(param.Position);
+        return Task.FromResult(Translator.ToLsp(this.signatureService.GetSignature(this.syntaxTree, this.semanticModel, cursorPosition)));
```

Comment:
Please factor out `this.signatureService.GetSignature(this.syntaxTree, this.semanticModel, cursorPosition)` into a local to make this easier to read